timemachine/potentials/nonbonded.py

```diff
@@ -251,6 +253,17 @@ def apply_cutoff(x):
     return vdW, electrostatics
 
 
+def nonbonded_v3_interaction_groups(conf, params, box, inds_l, inds_r, beta: float, cutoff: Optional[float] = None):
+    """Nonbonded interactions between all pairs of atoms $(i, j)$
+    where $i$ is in the first set and $j$ in the second.
+
+    See nonbonded_v3 docstring for more details
+    """
+    pairs = np.array(list(itertools.product(inds_l, inds_r)))
+    vdW, electrostatics = nonbonded_v3_on_specific_pairs(conf, params, box, pairs[:, 0], pairs[:, 1], beta, cutoff)
```

Comment:
Wow, good catch! Switched to using `np.meshgrid` in 91d1722 (seems slightly more helpful than `np.indices` in this case, since we're passing in arrays of indices) and also saw a ~5000x speedup.